nixos/modules/services/x11/display-managers/sddm.nix

```diff
@@ -110,22 +108,22 @@ in
         default = true;
         description = ''
           Whether to enable automatic HiDPI mode.
-          </para>
-          <para>
-          Versions up to 0.17 are broken so this only works from 0.18 onwards.
         '';
       };
 
-      extraConfig = mkOption {
-        type = types.lines;
-        default = "";
+      settings = mkOption {
+        type = types.attrs;
```

Comment:
We have some fancy pants new types available. Take a look at [this](https://github.com/NixOS/nixpkgs/blob/3307adf755a1f6262cd42e73e30ba58261cd868a/nixos/modules/tasks/filesystems/nfs.nix), specifically `format = pkgs.formats.ini {};`.

Reply:
Nice, thanks, fixed.